updates/P6.txt

```diff
@@ -0,0 +1,135 @@
+back. The two consecutive blocks must also be within the same epoch.
+
+Leaders are determined per epoch, and are known to everybody for the epoch.
+The notion of a payday, as it exists in protocols 4 and 5, remains largely
+unchanged, although payday transitions are slightly changed due to the
+technicalities of how epoch transitions occur in Concordium BFT.
+However in normal conditions there should be little observable effect on
+users of the chain.
+
+                               Changes
+
+The following behaviours are changed in protocol version 6.
+
+1. Concordium BFT takes effect. In particular this changes the format of
+   blocks, and the messages exchanged by the nodes.
+2. The election difficulty chain parameter is removed.
+3. There are new chain parameters
+   - the minimum timeout, and factors for increasing and decreasing round
+     timeout duration on successive round timeouts and finalizations,
+     respectively
+   - minimum block time, controlling minimum time between blocks in two
+     successive rounds
+   - minimum number of finalizers and maximum number of finalizers. These,
+     together with the existing finalizer relative stake threshold
+     parameter, revise the condition on being a finalizer. The relative
+     stake threshold is not interpreted relative to the total effective
+     stake of bakers. The minimum number of finalizers must be reached
+     before the relative stake threshold applies. Analogously, the number
+     of finalizers are capped.
+4. Block energy limit becomes an updatable parameter.
+5. Smart contract validation has the following changes
+   - Version 1 smart contracts can now use sign extension Wasm
+     instructions (i32.extend8_s, i32.extend16_s, i64.extend8_s,
+     i64.extend16_s, i64.extend32_s).
+   - In version 1 smart contracts, the contract schema, and other custom
+     sections, no longer count towards the cost when executing contracts
+     for modules that were deployed in protocol 6.
+   - The behaviour of rollbacks in certain cases of reentrant contract
+     calls is corrected to conform to the intended semantics. In protocols
+     4 and 5 state changes might not be rolled back in case of failure of
+     nested, reentrant, contract calls.
+6. In V1 smart contracts, the `invoke` operation allows for two additional
+   operations, querying account keys, and checking a signature with respect
+   to the account keys.
+
+                               Effects
+
+1. The consensus change has no direct effects on the state of accounts and
+   contracts. The overall immediate effect of the new consensus will be, in
+   normal network conditions, that transaction confirmation times will have
+   much less variance.
+
+2. Behaviour of existing V1 contract instances may be affected in the
+   following scenarios
+   - the contract attempted to perform an operation with tags 5 or 6 the
+     execution would have triggered a runtime error, whereas in protocol 6
```

Comment:
Grammar? Should it be "if the contract..."? Generally quite long sentence that is hard to follow. It's also unclear to me what these tags are.

Reply:
I revised a bit,

Tags are just operation ids. I don't want go into their details here, this is the level of details we used for previous documents.